File: TASC-Intern-Batch9/fe/pharmacy-app/src/api/rateService.ts
import axiosClient from './axiosClient';
import { ApiResponse, PageResponse } from '../types';
import { RateRequest, RateResponse } from '../types/rate';

const rateService = {
  createRate: (data: RateRequest) => {
    return axiosClient.post<ApiResponse<RateResponse>>('/rates', data);
  },

  getRatesByProduct: (
    productId: number,
    pageIndex: number = 1,
    pageSize: number = 10,
    rating?: number
  ) => {
    const params: any = { pageIndex, pageSize };
    if (rating) {
      params.rating = rating;
    }
    return axiosClient.get<ApiResponse<PageResponse<RateResponse[]>>>(`/rates/${productId}`, {
      params
    });
  }
};

export default rateService;

File: pharmacy-v1/pharmacy-backend-microservice/order-service/src/main/java/com/pharmacy_backend/order_service/service/RateService.java
package com.pharmacy_backend.order_service.service;

import com.pharmacy_backend.common.dto.response.ApiResponse;
import com.pharmacy_backend.common.dto.response.PageResponse;
import com.pharmacy_backend.order_service.dto.request.RateRequest;
import com.pharmacy_backend.order_service.dto.response.RateResponse;

import java.util.List;

public interface RateService {
    ApiResponse<PageResponse<List<RateResponse>>> getRatesByProductId(Long productId,
                                                            Integer pageIndex,
                                                            Integer pageSize, Integer rating);

    ApiResponse<Void> createRate(RateRequest request);
}
